Source/WTF/wtf/Algorithms.h

```diff
@@ -54,5 +58,40 @@ bool allOf(ContainerType&& container, AllOfFunction allOfFunction)
     return true;
 }
 
+template<typename T, typename U>
+void memcpySpan(std::span<T> destination, std::span<U> source)
+{
+    RELEASE_ASSERT(destination.size() == source.size());
+    static_assert(sizeof(T) == sizeof(U));
+    static_assert(!std::is_const_v<T>, "memcpySpan will not remove constness from destination");
+    memcpy(reinterpret_cast<void*>(destination.data()), reinterpret_cast<void*>(const_cast<std::remove_const_t<U>*>(source.data())), destination.size() * sizeof(T));
 }
 
+template<typename T>
+void memsetSpan(std::span<T> destination, uint8_t byte)
+{
+    static_assert(!std::is_const_v<T>, "memsetSpan will not remove constness from destination");
+    memset(reinterpret_cast<void*>(destination.data()), byte, destination.size() * sizeof(T));
+}
+
+template<typename T, typename U>
+std::span<T> spanCast(std::span<U> span)
+{
+    RELEASE_ASSERT(!(span.size_bytes() % sizeof(T)));
```

Comment:
Don't we want to static_assert() that T and U have the same size?

Reply:
> Don't we want to static_assert() that T and U have the same size?

No, that's the whole point of this method.  To cast from a `uint8_t` span to a span of a different type.  We just want to make sure that we don't have any leftover bytes.

Unless you would like a size (count) of `T` objects passed in to make sure that converting from `U` to `T` also has the expected number of `T` objects as well.

This is used in the `IPC::Decoder` class.